pkg/kvstore/allocator/allocator.go

```diff
@@ -208,9 +179,16 @@ type Allocator struct {
 	// backoffTemplate is the backoff configuration while allocating
 	backoffTemplate backoff.Exponential
 
-	// idWatcherStop is the channel used to stop the kvstore watcher
-	idWatcherStop chan struct{}
-	idWatcherWg   sync.WaitGroup
+	// mainCache is the main cache, representing the allocator contents of
+	// the primary kvstore connection
+	mainCache cache
+
+	// remoteCachesMutex protects accesse to remoteCaches
+	remoteCachesMutex lock.RWMutex
+
+	// remoteCaches is the list of additional remote caches being watched
+	// in addition to the main cache
+	remoteCaches map[*RemoteCache]struct{}
```

Comment:
Having pointers as a map key will have random behavior.

Reply:
It doesn't in this case. It's handled identically as if we would convert the pointer value to a string and use it as an ID.

Reply:
Correct.